libs/liquidity/src/lib/liquidity-data-provider.ts

```diff
@@ -76,6 +78,31 @@ export const liquidityProvisionsDataProvider = makeDataProvider<
   },
 });
 
+function isLpProvision(
```

Comment:
suggestion (non blocking): I think the naming here is off. They are both LPs but one is a subscription update and one is a query result. Perhaps `isLPFragment` would be better?

Reply:
Secondly the function body is super hard to read, can we simplify to the following. The type discrimination on the __typename field should give us all the type safety we need

```
function isLPFragment(
  entry:
    | LiquidityProvisionFieldsFragment
    | IterableElement<
        LiquidityProvisionsUpdateSubscription['liquidityProvisions']
      >
): entry is LiquidityProvisionFieldsFragment {
  if (entry.__typename === 'LiquidityProvision') {
    return true;
  }
  return false;
}
```

Reply:
agree, updated in last commit.